src/externs.ts

```diff
@@ -0,0 +1,439 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * @fileoverview Externs creates Closure Compiler \@externs definitions from the ambient
+ * declarations in a TypeScript file.
+ *
+ * For example, a
+ *   declare interface Foo { bar: string; }
+ * Would generate a
+ *   /.. \@externs ./
+ *   /.. \@record ./
+ *   var Foo = function() {};
+ *   /.. \@type {string} ./
+ *   Foo.prototype.bar;
+ *
+ * The generated externs indicate to Closure Compiler that symbols are extern to the optimization
+ * process, i.e. they are provided by outside code. That most importantly means they must not be
+ * renamed or removed.
+ */
+
+import * as path from 'path';
+
+import * as jsdoc from './jsdoc';
+import {AnnotatorHost, escapeForComment, isValidClosurePropertyName, maybeAddHeritageClauses} from './jsdoc_transformer';
+import {ModuleTypeTranslator} from './module_type_translator';
+import {getEntityNameText, getIdentifierText, hasModifierFlag, isDtsFileName} from './transformer_util';
+import * as ts from './typescript';
+
+/**
+ * Symbols that are already declared as externs in Closure, that should
+ * be avoided by tsickle's "declare ..." => externs.js conversion.
+ */
+const CLOSURE_EXTERNS_BLACKLIST: ReadonlyArray<string> = [
+  'exports',
+  'global',
+  'module',
+  // ErrorConstructor is the interface of the Error object itself.
+  // tsickle detects that this is part of the TypeScript standard library
+  // and assumes it's part of the Closure standard library, but this
+  // assumption is wrong for ErrorConstructor.  To properly handle this
+  // we'd somehow need to map methods defined on the ErrorConstructor
+  // interface into properties on Closure's Error object, but for now it's
+  // simpler to just blacklist it.
+  'ErrorConstructor',
+  'Symbol',
+  'WorkerGlobalScope',
+];
+
+/**
+ * generateExterns generates extern definitions for all ambient declarations in the given source
+ * file. It returns a string representation of the Closure JavaScript, not including the externs
```

Comment:
Not for this CL, but do you think it's eventually worth using transformers here too?